packages/cli/pipelines/from-s3.ttl

```diff
@@ -18,3 +18,22 @@ _:openCsvFromBucket
                        "s3endpoint"^^:VariableName
                        "s3bucket"^^:VariableName
                      ) .
+
+<#LoadDatacubeMetadata>
+  a :Pipeline, :ReadableObjectMode ;
+  :steps [ :stepList ( <#loadDatacubeMetadata> ) ] ;
+  :variables [ :variable
+    [ a :Variable; :name "s3endpoint" ] ,
+    [ a :Variable; :name "s3bucket" ]
+  ].
+
+<#loadDatacubeMetadata>
+  a :Step ;
+  code:implementedBy [
+    code:link <file:../lib/datacube-metadata#openFileFromS3> ;
```

Comment:
Anything preventing sharing the code which opens the input file streams?

Reply:
The reason I didn't reuse the code is that I need to handle the case where the file doesn't exist and haven't found a way to do this with a pipeline step. But if you have an idea, I'm open ;)

Reply:
Not sure. The code of `lib/datacube-metadata#openFileFromS3` and `lib/s3#openFile` look fairly similar to me

Reply:
The difference is that `lib/s3#openFile` will crash the pipeline if the file doesn't exist (which is what we want), whereas `lib/datacube-metadata#openFileFromS3` will return a stream with an empty content.

Reply:
Still pretty close in terms of actually using the S3 API.

I'm not insisting. But the third time... ;)

Reply:
What do you suggest?

Reply:
Nothing at this moment. If we find ourselves writing similar code again, then we should consider the possibility that we only need one copy

Reply:
I tried to extract the common parts in a function... not sure it was worth it though.